torch/lib/c10d/Utils.cpp

```diff
@@ -185,7 +186,16 @@ int connect(
       pfd.fd = socket;
       pfd.events = POLLOUT;
 
-      int numReady = ::poll(&pfd, 1, timeout.count());
+      int pollTimeout = timeout.count();
```

Comment:
It reads a bit weird that you first set `pollTimeout` and then check the other variable. Could you instead assign `pollTimeout = -1` to be clear that this is the default? This makes it clear(er) IMO that it'll have a negative value if `timeout == kNoTimeout`. As is, we could redefine `kNoTimeout` to be 0 and not be aware of any harm.